course3/homework-unique_ptr/test.cc

```diff
@@ -0,0 +1,136 @@
+#include "unique_ptr.h"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+
+
+struct Number
+{
+	int n;
+	static int ctr;
+	Number(int n) : n(n) { ctr++; }
+	Number(int n1, int n2) : n(n1 + n2) { ctr++; }
+	Number(const Number& n) : n(n.n) { ctr++; }
+	Number(Number& n) : n(n.n * 10) { ctr++; }
+	~Number() { ctr--; }
+};
+int Number::ctr = 0;
+
+void unique_ptr_test()
```

Comment:
Got rid of the fancy template, which makes it much easier to write tests. Would still like an easy way to quickly switch between std::unique_ptr and karun::unique_ptr when writing new tests in order to ensure their behaviour is consistent. Is it okay to do something like `using UniquePtr = karun::unique_ptr<Number>` as the first line in unique_ptr_test, so I just have to change one line if I want to try std::unique_ptr instead? If so, can I do something similar for karun::make_unique (`using MakeUnique<number> = karun::make_unique` doesn't work since make_unique is not a type)?